client/components/account-balances/balance-block.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { formatCurrency } from 'wcpay/utils/currency';
+import Loadable from 'components/loadable';
+import { fundLabelStrings } from './strings';
+
+/**
+ * balanceType
+ */
+type balanceType = 'pending' | 'available' | 'reserved';
+
+/**
+ * BalanceBlockProps
+ *
+ * @typedef {Object} BalanceBlockProps
+ *
+ * @property {balanceType} type    Type of the balance block. eg "available".
+ * @property {string} currencyCode Currency code of the balance block.
+ * @property {number} [amount]     Optional. The balance amount.
+ * @property {boolean} [isLoading] Optional. Whether the balance block is loading.
+ */
+interface BalanceBlockProps {
+	type: balanceType;
+	currencyCode: string;
+	amount?: number;
+	isLoading?: boolean;
+}
+
+/**
+ * Renders a balance block including a title and amount.
+ *
+ * @param {BalanceBlockProps} props   Balance block props. See `BalanceBlockProps` interface.
+ * @param {balanceType} props.type    The balance type. eg "available".
+ * @param {string} props.currencyCode Currency code of the balance block.
+ * @param {number} [props.amount]     Optional. The balance amount.
+ * @param {boolean} [props.isLoading] Optional. Whether the balance block is loading.
+ *
+ * @return {JSX.Element} Rendered balance element.
+ */
+const BalanceBlock: React.FC< BalanceBlockProps > = ( {
+	type,
+	currencyCode,
+	amount = 0,
+	isLoading = false,
+} ) => {
+	return (
+		<div
+			className={ `wcpay-account-balances__balances__item wcpay-account-${ type }__balance_item` }
+		>
+			<p className="wcpay-account-balances__balances__item__title">
+				{ fundLabelStrings[ type ] }
+			</p>
+			<p className="wcpay-account-balances__balances__item__amount">
```

Comment:
Nit: we could add a title to improve accessibility and allow for easier component testing. Even better, we could use an attribute like `aria-labelledby` to refer to the `p` element that holds this title already.
```suggestion
			<p
				className="wcpay-account-balances__balances__item__title"
				title="fundLabelStrings[ type ]"
			>
```

Reply:
I ended up going with the `aria-labelledby` approach but that required generating an `id`. I don't know if I did that correctly.

Reply:
These look good to me!

I double-checked with [Axe devtools](https://github.com/dequelabs/axe-core) as well ✅

Reply:
Nice!